### PR TITLE
Kafka publisher

### PR DESCRIPTION
**Pull request by pdeziel**
This adds a Kafka publisher to Baleen which currently supports publishing messages to two topics: feeds and documents. I also added a configuration module to support reading in configuration parameters from the environment.

### CODE REVIEW

File: publish/kafka.go [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+		writer.Balancer = &kafka.LeastBytes{}
+	case "Hash":
+		writer.Balancer = &kafka.Hash{}
+	case "Murmur2":
+		writer.Balancer = &kafka.Murmur2Balancer{}
+	case "CRC32":
+		writer.Balancer = &kafka.CRC32Balancer{}
+	default:
+		return nil, fmt.Errorf("unknown balancer specified: %s", conf.Balancer)
+	}
```
Comment (rebeccabilbro) [inline publish/kafka.go:38]: Neat! Is this sort of boilerplate for Kafka or something custom?

File: store/store.go [hunk 2]
```diff
@@ -36,6 +36,12 @@ type Document struct {
 	Link         string
 }
 
+type Feed struct {
+	URL    string `json:"url"`
+	Active bool   `json:"active"`
```
Comment (rebeccabilbro) [inline store/store.go:41]: I think this `Active` field will be very useful!

**Review by rebeccabilbro — APPROVED** [review]
This looks great @pdeziel! Adding in the event stream is going to really augment what we can do with Baleen. Already it seems really useful to be able to think about things in terms of documents vs feeds -- e.g. things that are getting consumed by orca for preprocessing/modeling vs consumed by the frontend for diagnostic purposes (is this feed still on?). I also really like what you've done with the config package; it's a definitive move away from the toy project that this started out as!

File: publish/kafka.go (hunk 1 above)
Comment (pdeziel) [reply publish/kafka.go:38]: Basically trying to understand what options we have for balancers beyond the default.

**Pull request merged**